crates/factor-outbound-networking/src/lib.rs

```diff
@@ -157,11 +157,19 @@ impl FactorInstanceBuilder for InstanceBuilder {
 }
 
 // TODO: Refactor w/ spin-outbound-networking crate to simplify
+#[derive(Clone)]
 pub struct OutboundAllowedHosts {
     allowed_hosts_future: SharedFutureResult<AllowedHostsConfig>,
 }
 
 impl OutboundAllowedHosts {
+    pub async fn resolve(&self) -> anyhow::Result<Arc<AllowedHostsConfig>> {
+        self.allowed_hosts_future.clone().await.map_err(|err| {
+            // TODO: better way to handle this?
```

Comment:
I know this was just moved, but why doesn't `?` work?

Reply:
`err` is an `Arc<Error>` here